### PR TITLE
[Feature] Create HiddenUi via prefab field

### PR DESCRIPTION
**Pull request by Telzhaak**
Aims to make it possible to hide parts or the complete Ui of a prefab,
both via the prefab, and the `UiCreator::create()` function.

Current usage:
```ron
transform:(
// snip
hidden: true,
)
```
This adds a `HiddenPropagate` component to this entity. When using the `HideHierarchySystem` this allows hiding and unhiding of every entity below it at run-time (which would not be possible with only a `Hidden` component. It is still advised to set this variable for EVERY child entity of the entity you hide, otherwise the child entities might be drawn once before the `HideHierarchySystem` kicks in (depends on the order of dependencies)

The `create` function is work in progress.

### CODE REVIEW

**Comment by Telzhaak** [comment]
I admit defeat.
No idea how to add a component from the `Loader.load()` function...

It works from inside a prefab at least.

**Comment by jojolepro** [comment]
> No idea how to add a component from the Loader.load() function...
You are not supposed to. This returns a handle to the stored asset in AssetStorage<T>

**Comment by Telzhaak** [comment]
I may not be supposed to, but i'd still want to be able to load Ui in a hidden state and only make it visible once it is needed.

For now the prefab-method will suffice

File: amethyst_ui/src/prefab.rs [hunk 1]
```diff
@@ -83,6 +85,12 @@ impl UiTransformBuilder {
         self
     }
 
+    /// Set to event transparent
```
Comment (Moxinilian) [inline amethyst_ui/src/prefab.rs]: What does that mean?

**Comment by Moxinilian** [comment]
That's what prefabs are for by the way, the loader only loads assets with no context, just plain data, when a prefab loads assets and context in the world.

File: amethyst_ui/src/prefab.rs (hunk 1 above)
Comment (Telzhaak) [reply amethyst_ui/src/prefab.rs]: I have no idea

**Comment by Telzhaak** [comment]
No idea why rustfmt is complaining though.

I'm on latest stable (just updated) and nothing is changed by running `cargo fmt`

**Pull request merged**